src/services/endpoint/EndpointService.ts

```diff
@@ -37,12 +38,19 @@ export class EndpointService {
     this.dmrService.readResource(request)
     .then((endpointResponse: any): IEndpoint[] => {
       let endpoints: IEndpoint[] = [];
-      for (let name in endpointResponse) {
-        if (isNotNullOrUndefined(endpointResponse[name])) {
-          let endpointObject: any = endpointResponse[name][name];
-          endpoints.push(EndpointService.parseEndpoint(name, endpointObject));
+      traverseObjectRecursively(endpointResponse, (visitedObject: any) => {
+        let isEndpoint: boolean = isNotNullOrUndefined(visitedObject["cache-container"]);
+        let isMultiRouterEndpoint: boolean = isNotNullOrUndefined(visitedObject["rest-socket-binding"])
+          || isNotNullOrUndefined(visitedObject["hotrod-socket-binding"]);
+        if (isEndpoint) {
+          let endpoint: IEndpoint = EndpointService.parseEndpoint(visitedObject.name, visitedObject);
+          if (endpoint["cache-container"] === cacheContainer.name) {
+            endpoints.push(endpoint);
+          }
+        } else if (isMultiRouterEndpoint) {
+          //todo parse multi-tenant router
```

Comment:
Is there anything stopping us catering for mult-tenant routers now? If not, I think it makes sense for us to get this right before we integrate.